extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -351,6 +358,49 @@ public function getWikiVariables() {
 	}
 
 	/**
+	 * @desc for classic or CK editor markup return
+	 * wikitext ready to process and display in Mercury skin
+	 *
+	 * @throws \BadRequestApiException
+	 * @throws \MWException
+	 */
+	public function getArticleFromMarkup() {
+		global $wgUser, $wgRequest;
+
+		if ( !$wgRequest->wasPosted() ) {
```

Comment:
shouldn't we also check for `editToken`? if yes consider using `checkWriteRequest`
https://github.com/Wikia/app/blob/ccb80e110026e9407ae902d90b9e022be8f20994/extensions/wikia/UserLogin/UserLoginSpecialController.class.php#L654